File: app/src/main/java/com/example/purepawapp/ui/blog/BlogListFragment.java
package com.example.purepawapp.ui.blog;

import android.os.Bundle;
import android.view.View;

import androidx.navigation.fragment.NavHostFragment;

import com.example.purepawapp.R;
import com.example.purepawapp.databinding.FragmentBlogListBinding;
import com.example.purepawapp.ui.common.BaseFragment;
import com.google.android.material.card.MaterialCardView;

import java.util.LinkedHashMap;
import java.util.Map;

public class BlogListFragment extends BaseFragment<FragmentBlogListBinding> {

    public BlogListFragment() {
        super(FragmentBlogListBinding::inflate);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        getBinding().btnBack.setOnClickListener(v -> NavHostFragment.findNavController(this).popBackStack());

        Map<MaterialCardView, String> cardToBlogId = new LinkedHashMap<>();
        cardToBlogId.put(getBinding().cardBlog1, "blog-1");
        cardToBlogId.put(getBinding().cardBlog2, "blog-2");
        cardToBlogId.put(getBinding().cardBlog3, "blog-3");
        cardToBlogId.put(getBinding().cardBlog4, "blog-4");
        cardToBlogId.put(getBinding().cardBlog5, "blog-5");

        for (var entry : cardToBlogId.entrySet()) {
            entry.getKey().setOnClickListener(v -> {
                Bundle args = new Bundle();
                args.putString("blogId", entry.getValue());
                NavHostFragment.findNavController(this).navigate(R.id.action_blogListFragment_to_blogDetailFragment, args);
            });
        }
    }
}

File: app/src/main/java/com/example/purepawapp/ui/blog/BlogListFragment.kt
package com.example.purepawapp.ui.blog

import android.os.Bundle
import android.view.View
import androidx.core.os.bundleOf
import androidx.navigation.fragment.findNavController
import com.example.purepawapp.R
import com.example.purepawapp.databinding.FragmentBlogListBinding
import com.example.purepawapp.ui.common.BaseFragment

class BlogListFragment : BaseFragment<FragmentBlogListBinding>(FragmentBlogListBinding::inflate) {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.btnBack.setOnClickListener {
            findNavController().popBackStack()
        }

        val cardToBlogId = mapOf(
            binding.cardBlog1 to "blog-1",
            binding.cardBlog2 to "blog-2",
            binding.cardBlog3 to "blog-3",
            binding.cardBlog4 to "blog-4",
            binding.cardBlog5 to "blog-5"
        )

        cardToBlogId.forEach { (card, blogId) ->
            card.setOnClickListener {
                findNavController().navigate(
                    R.id.action_blogListFragment_to_blogDetailFragment,
                    bundleOf("blogId" to blogId)
                )
            }
        }
    }
}
